docs/hugo/content/design/ADR-2023-09-Skipping-Properties/_index.md

```diff
@@ -0,0 +1,418 @@
+
+When converting a resource to an earlier version, properties introduced in that later version get stored in the `PropertyBag` on the earlier version to ensure they're not lost.
+
+Similarly, when converting a resource to a later version, properties removed from that later version get stored in the `PropertyBag` of the later version to ensure they are not lost.
+
+
+### Existing Imports
+
+Go prohibits (for good reason) import loops between packages. We already have consistent imports from earlier package versions to later ones. These prohibit us adding additional imports in the other direction.
+
+### Option 1: Add Synthetic Property
+
+Standardize on storing a **v3** `Address` in the property bag, even when the resource is a **v5** `Person`, by injecting a new (synthetic) property into the **v4** `Person` with an identical shape (aka serialization format) to the **v3** `Address`.
+
+<!-- yuml.me
+
+[v3.Person|FullName string;FamilyName string; KnownAs string; ResidentialAddress v3.Address|Properties PropertyBag]
+[v3.Person]->[v3.Address|Label multilineString|Properties PropertyBag]
+
+[v4.Person|FullName string;FamilyName string; KnownAs string;ResidentialAddress v4.Address|Properties PropertyBag]
+[v4.Person]->[v4.Address|Label multilineString|Properties PropertyBag]
+
+[v5.Person|FullName string;FamilyName string; KnownAs string;ResidentialAddress v5.Address|Properties PropertyBag]
+[v5.Person]->[v5.Address|Street string; Suburb string; City string; Country string|Properties PropertyBag]
+-->
+
+{{< figure src="./option-1-class.png" >}}
+
+The shape of `Address` when reintroduced will always match the shape of `Address` when removed, so we can always serialize the same shape into or outof the property bag, ensuring any `Address` has the a consistent shape.
+
+Pros:
+* Works the way existing conversions do.
+* We have an existing contract for augmenting conversions with custom code, allowing us to handcode the conversion between **v4** and **v5** to handle the mismatch in the shapes of `Address`.
+* Addresses the central inconsistency problem, ensuring we always serialize the same shape into intermediate property bags
+
+Cons:
+* Changes the serialization format for existing **v4** resources. To avoid a breaking change, we'd need to code generate conversions for several edge cases, including:
+  * Conversion of a _new style_ **v4** `Person` to **v3** person, handling both the case where the address is in the property bag, and the case where it is a property reference.
+  * Conversion of a _new style_ **v4** `Person` to **v5** person, handling both the case where the address is in the property bag (requiring conversion from the **v3** shape), and the case where it is a property reference (requiring conversion from the **v4** shape).
+* Changes the object model of released resources, a breaking change for code consumers of the object model.
```

Comment:
Another con is that it seems to be an end-run around `PropertyBag`? If we took this approach, would we be using PropertyBag for anything?  I guess it would still be used for v5 adds a property v4 doesn't have, then when converting v5->v4 that property goes into the property bag?

But when going forward (v3 -> v4), if v4 removed a property that v3 had, then it would be resurrected as a property on v4 anyway?

I think it ends up adding more confusion to an already somewhat complex topic, because no longer is the answer "it's always in the propertybag" now it depends.